src/AlarmClock.h

```diff
@@ -1,37 +1,47 @@
 /* 
  * File:   AlarmClock.h
- * Author: Craig Cogdill
- * Created: October 15, 2015 10:45am
+ * Author: Amanda Carbonari
+ * Created: January 5, 2016 4:35pm
+ * Description: Uses an "interruptable" std::thread, that checks every
+ *    microsecond to see if any of the two interrupt atomics have been set to
+ *    true (mReset, mExit). If they are it will stop it's sleep and then check
+ *    to see if it was a reset or exit. If it was expired and not reset or exit it will sleep
+ *    until mReset (or exit) is set to true and it can start sleeping again.  
  */
 
 #pragma once
 #include <chrono>
 #include <thread>
 #include <atomic>
-#include <future>
 #include <functional>
+#include <iostream>
 #include "StopWatch.h"
-
+using namespace std;
 
 template<typename Duration> class AlarmClock {
 public:
-   typedef std::chrono::microseconds microseconds;
-   typedef std::chrono::milliseconds milliseconds;
-   typedef std::chrono::seconds seconds;
-   
-   AlarmClock(unsigned int sleepDuration, std::function<void(unsigned int)> funcPtr = nullptr) : mExpired(false),
-      kSleepTime(sleepDuration),
-      kSleepTimeMsCount(ConvertToMillisecondsCount(Duration(sleepDuration))),
+   typedef chrono::microseconds microseconds;
+
+   // The sleep function is passed in for the unit tests. 
+   AlarmClock(unsigned int sleepDuration, function<bool (unsigned int)> funcPtr = nullptr) : mExpired(false),
+      mExit(false),
+      mReset(false),
       kSleepTimeUsCount(ConvertToMicrosecondsCount(Duration(sleepDuration))),
-      mSleepFunction(funcPtr) {
-         if (mSleepFunction == nullptr) {
-            mSleepFunction = std::bind(&AlarmClock::SleepUs, this, std::placeholders::_1);
+      mAlarmExpiredFunction(funcPtr) {
+         if (mAlarmExpiredFunction == nullptr) {
+            mAlarmExpiredFunction = [&](unsigned int sleepTime) -> bool
```

Comment:
@acarb95 this might seem cryptic to people.  Can you please put in an explanation why the lambda is needed so it's there for when we next time need to update this code